compiler/ilgen/OMRIlBuilder.cpp

```diff
@@ -2545,26 +2545,111 @@ OMR::IlBuilder::Switch(const char *selectionVar,
    TR::Node *lookupNode = TR::Node::create(TR::lookup, numCases + 2, loadValue(selectorValue), defaultNode);
 
+   generateSwitchCases(lookupNode, defaultNode, defaultBuilder, numCases, cases);
+   }
+
+void
+OMR::IlBuilder::Switch(const char *selectionVar,
+                  TR::IlBuilder **defaultBuilder,
+                  uint32_t numCases,
+                  ...)
+   {
+   va_list args;
+   va_start(args, numCases);
+   JBCase **cases = createCaseArray(numCases, args);
+   va_end(args);
+
+   Switch(selectionVar, defaultBuilder, numCases, cases);
+   }
+
+void
+OMR::IlBuilder::TableSwitch(const char *selectionVar,
+                  TR::IlBuilder **defaultBuilder,
+                  bool generateBoundsCheck,
+                  uint32_t numCases,
+                  JBCase **cases)
+   {
+   TR::IlValue *selectorValue = Load(selectionVar);
+   TR_ASSERT(selectorValue->getDataType() == TR::Int32, "TableSwitch only supports selector having type Int32");
+   if (numCases > 0)
+      {
+      int32_t low = cases[0]->_value;
+      int32_t high = cases[numCases - 1]->_value;
+      if (low != 0)
+         selectorValue = Sub(selectorValue, ConstInt32(low));
+      }
+
+   *defaultBuilder = createBuilderIfNeeded(*defaultBuilder);
+
+   TR::Node *defaultNode = TR::Node::createCase(0, (*defaultBuilder)->getEntry()->getEntry());
+   TR::Node *tableNode = TR::Node::create(TR::table, numCases + 2, loadValue(selectorValue), defaultNode);
+   if (!generateBoundsCheck)
+       tableNode->setIsSafeToSkipTableBoundCheck(true);
+
+   generateSwitchCases(tableNode, defaultNode, defaultBuilder, numCases, cases);
+   }
+
+void
+OMR::IlBuilder::TableSwitch(const char *selectionVar,
+                  TR::IlBuilder **defaultBuilder,
+                  bool generateBoundsCheck,
+                  uint32_t numCases,
+                  ...)
+   {
+   va_list args;
+   va_start(args, numCases);
+   JBCase **cases = createCaseArray(numCases, args);
+   va_end(args);
+
+   TableSwitch(selectionVar, defaultBuilder, generateBoundsCheck, numCases, cases);
+   }
+
+TR::IlBuilder::JBCase *
+OMR::IlBuilder::MakeCase(int32_t caseValue, TR::IlBuilder **caseBuilder, int32_t caseFallsThrough)
+   {
+   TR_ASSERT(caseBuilder != NULL, "MakeCase, needs to have non-null caseBuilder");
+   *caseBuilder = createBuilderIfNeeded(*caseBuilder);
+   auto * c = new (_comp->trHeapMemory()) JBCase(caseValue, *caseBuilder, caseFallsThrough);
+   return c;
+   }
+
+TR::IlBuilder::JBCase **
+OMR::IlBuilder::createCaseArray(uint32_t numCases, va_list args)
+   {
+   JBCase **cases = (JBCase **) _comp->trMemory()->allocateHeapMemory(numCases * sizeof(JBCase *));
+   TR_ASSERT(NULL != cases, "out of memory");
+
+   for (uint32_t c = 0; c < numCases; ++c)
+      {
+      cases[c] = va_arg(args, JBCase *);
+      }
+
+   return cases;
+   }
+
+void
+OMR::IlBuilder::generateSwitchCases(TR::Node *switcheNode, TR::Node *defaultNode, TR::IlBuilder **defaultBuilder, uint32_t numCases, JBCase **cases)
```

Comment:
is this a typo? Should be `switchNode` maybe?

Reply:
Corrected